### PR TITLE
Add some ignoreUsedDependencies that are needed in the build

### PR DESCRIPTION
**Pull request by cunningt**
Add some ignoreUsedDependencies that are needed in the build, and add a plugin repository statement for SNAPSHOTS because the syndesis-build-helper-maven-plugin is used and I'm getting errors when running mvn clean and it can't resolve the plugin.

### CODE REVIEW

**Comment by cunningt** [comment]
@KurtStam I redid the request from https://github.com/syndesisio/syndesis/pull/639 here - I think we do need both the pluginRepo here and the ignoreUnused statements.

**Comment by KurtStam** [comment]
@cunningt thx Tom, can you rebase once more to resolve the conflict?

File: app/rest/runtime/pom.xml [hunk 1]
```diff
@@ -48,6 +48,18 @@
...
+      <id>sonatype-snapshot</id>
+      <name>Sonatype snapshot repository</name>
+      <url>https://oss.sonatype.org/service/local/repositories/snapshots/content</url>
+      <releases>
+        <enabled>false</enabled>
+      </releases>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+    </pluginRepository>
```
Comment (rhuss) [inline app/rest/runtime/pom.xml]: Where do we need snapshot plugin form Maven central ?

File: app/verifier/pom.xml [hunk 2]
```diff
@@ -365,6 +365,10 @@
         <artifactId>maven-dependency-plugin</artifactId>
         <configuration>
           <ignoredUnusedDeclaredDependencies>
+            <ignoredUnusedDeclaredDependency>io.syndesis:sql-common</ignoredUnusedDeclaredDependency>
+            <ignoredUnusedDeclaredDependency>junit:junit</ignoredUnusedDeclaredDependency>
+            <ignoredUnusedDeclaredDependency>commons-io:commons-io</ignoredUnusedDeclaredDependency>
+            <ignoredUnusedDeclaredDependency>org.assertj:assertj-core</ignoredUnusedDeclaredDependency>
```
Comment (rhuss) [inline app/verifier/pom.xml]: Sure that this needs to be added here and not by adding the proper scope to dependencies ? The maven dependency plugin does not count "test" and "runtime" scoped dependencies as unused, and e.g. "junit" should be clearly a test scoped dependency.

**Comment by KurtStam** [comment]
@cunningt I have approved but it needs another rebase to pick up my fix for the broken build.

**Pull request merged**